### PR TITLE
[Logo customization] Submit flow

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

added submit flow
added reset flow
reducers refactor

## Test

- launch admin with [menu-logo/submit](https://github.com/strapi/strapi/tree/menu-logo/submit) branch 
- launch getstarted with [menu-logo/wip-get-project-settings](https://github.com/strapi/strapi/tree/menu-logo/wip-get-project-settings) branch
- test changing menu logo
- test resetting menu logo

### CODE REVIEW

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/index.js [hunk 1]
```diff
@@ -16,26 +18,48 @@ import { Link } from '@strapi/design-system/Link';
...
-    // to do - handle this data
-    // const data = inputsRef.current.getValues();
+  const submitMutation = useMutation(body => postProjectSettings(body), {
+    onSuccess: async ({ menuLogo }) => {
+      await queryClient.invalidateQueries('project-settings', { refetchActive: true });
+      updateProjectSettings({ menuLogo: menuLogo?.url });
+    },
+  });
+
+  const handleSubmit = async () => {
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/index.js:51]: If this function is `async`, don't you need to `await` something? 🤔

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/Form/reducer.js [hunk 2]
```diff
@@ -8,14 +8,29 @@
 import produce from 'immer';
 
 const initialState = {
-  menuLogo: null,
+  menuLogo: {
+    display: null,
+    submit: {
+      rawFile: null,
+      isReset: false,
```
[the remark this answers is not in the document]
Comment (vincentbpro) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/Form/reducer.js:15]: It is used to know if a previously uploaded custom logo should be deleted on submit (and we come back to the default logo)
- To delete it we need to POST the `menuLogo` field set to `null` to the API
- This is diferent from submitting without changing the previous custom logo (in this case we do not want to POST a `menuLogo` field at all)
This `isReset` flag allows to make the difference between the two situations [see here](https://github.com/strapi/strapi/pull/13131/files#diff-b999175926cfe4662eb9c817575b9cdfcc388006b52f153ffa68b19ff4477d95R9)

However if you have a better naming suggestion we are open to it @HichamELBSI  :)

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/index.js (hunk 1 above)
Comment (vincentbpro) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/index.js]: You are right, nothing `async` called in here, removed the `await` ;)

**Review by HichamELBSI — APPROVED** [review]
LGTM

**Pull request merged**